jupyter_accessibility/dodo.py

```diff
@@ -46,87 +33,75 @@ def remove_directory(*dir):
-        main = cls(**ns.__dict__)
-
-        main.main(args, standalone=standalone)
-
-    @classmethod
-    def get_parser(cls):
-        """the argument parser for the Main cli"""
-        from argparse import ArgumentParser
-
-        parser = ArgumentParser("builder")
-        parser.add_argument("-i", "--ids", default=["jupyterlab"], help="repository ids", nargs="*")
-        parser.add_argument(
-            "-d", "--dir", default=Path("jupyter-ally-build"), help="the build directory", type=Path
-        )
-        return parser
-
-    @classmethod
-    def parse_args(cls, args=None):
-        parser = cls.get_parser()
-        return parser.parse_known_args(args=prep_args(args))
-
-
-class Tasks(Base):
-    def doit(self):
-        """build the doit application by pulling tasks from the"""
-        from doit.cmd_base import ModuleTaskLoader
-        from doit.doit_cmd import DoitMain
-
-        tasks = dict((x, getattr(self, x)) for x in dir(self) if x.startswith("task_"))
-        tasks.update({"DOIT_CONFIG": DOIT_CONFIG})
-        return DoitMain(ModuleTaskLoader(tasks))
-
-    def commands(self):
-        """explore what doit knows"""
-        return self.doit().get_commands()
-
-    def do(self, x, **kwargs):
-        """create a cmd action that can be run from the api"""
-        return do(f"{self.conda} {x}", **kwargs)
-
-
-def prep_args(x):
-    if isinstance(x, str):
-        from shlex import split
-
-        return split(x)
-    return x
+##############################################################################################
+## Tasks
+##############################################################################################
+
+
+build_dir = ROOT / "build"
+conda_env_dir = build_dir / ".env"
+prefix = conda_env_dir.resolve()
+conda = f"conda run --no-capture-output --prefix {prefix}"
+
+def task_create_env():
+    """Create a test environment into which dependencies needed by the test code will be installed"""
+    yield dict(
+        name="conda",
+        actions=[
+            do(
+                f'conda create -yc conda-forge --prefix {prefix} python=3.9 "nodejs>=14.18,<15" yarn git'
+            ),
+            do(f"{conda} python -m pip install pip --upgrade"),
+        ],
+        uptodate=[prefix.exists()],
+        clean=[(remove_directory, [prefix])],
+    )
+
+name = "jupyterlab"
+
+def task_install_app():
+    """Install the app under test into the test environment"""
+    yield dict(
+        name=f"pip:{name}",
+        actions=[do(f"{conda} python -m pip install {name}")],
+        clean=[do(f"{conda} python -m pip uninstall -y {name}")],
+    )
+
+def task_install_test_deps():
+    """Install the Python and Node.js dependencies needed by the test code"""
+    target = build_dir / AXE_TEMPLATE.name
+    yield dict(
+        name="copy-templates",
+        actions=[(move_directory, [AXE_TEMPLATE, target])],
+        clean=[(remove_directory, [target])],
+        uptodate=[target.exists()],
```

Comment:
naive question - but why do we need to move templates/files over?

Reply:
I don't think we do, but I wanted to keep the code execution as similar as possible to the way it was working before.